crates/driver/src/domain/competition/bad_tokens/mod.rs

```diff
@@ -102,6 +102,10 @@ impl Detector {
 
         if let Some(detector) = &self.simulation_detector {
             if let Some(quality) = detector.get_quality(token, now) {
+                if let Some(metrics) = &self.metrics {
+                    metrics.store_quality(token, quality);
+                }
```

Comment:
I think there was a misunderstanding. The bad token detection based on metrics / statistics should run without the need for the simulation based bad token detection.

Instead it should count how often a solution [fails to encode](https://github.com/cowprotocol/services/blob/main/crates/driver/src/domain/competition/mod.rs#L136) (simulation happens under the hood). That way the driver "learns" about the capabilities of the solver just by looking at its solutions.
This technically leaves a little wiggle room for false positives (tokens that only work because the settlement contract has enough fees buffered) but overall this should ™️ give the driver a pretty good indication whether or not the solver actually understands how to settle a specific token.

Conceptually it does exactly the same thing oncall does ([tutorial](https://www.notion.so/cownation/Debugging-Tutorials-68b6d8d39da94b1093a0032a273abeb7#e624a45cf9f54f659a8e3829f701527d) whenever we got an alert about driver errors. Just automatically so oncall no longer actually has to do it.

Reply:
My bad. I missed that `todo` 🙈 Updated the PR.